File: src/python/tsv/Compat/tsv_record.py
"""TsvRecord — production facade for tsv:record.

Spec authority: tsv:record
Fact ref: FACT-TSV-001
Canonical spec class: src/python/tsv/spec/record/record.py::Record
"""
from __future__ import annotations

from ..spec.record.record import Record as _SpecRecord


class TsvRecord(_SpecRecord):
    """Production facade for tsv:record."""

    spec_qname = "tsv:record"
    spec_fact_ref = "FACT-TSV-001"
    namespace_uri = "urn:iana:media-type:text:tab-separated-values"

File: src/net/tsv/Spec/TsvRecord.cs
// FormatFactory.Tsv — Spec.TsvRecord — Canonical spec-shaped model class
// spec_qname: tsv:record
// TC-QHARD-052: new canonical spec authority class for TSV format
namespace FormatFactory.Tsv.Spec;

/// <summary>
/// Canonical spec-shaped model class for a TSV record (data row).
///
/// A TSV record is a sequence of tab-separated fields (IANA text/tab-separated-values).
/// Each record corresponds to a row in the TSV document.
/// spec_qname: tsv:record
///
/// This is a canonical class in the Spec/ hierarchy. The facade wrapper is
/// FormatFactory.Tsv.TsvDocument (rows are exposed as string[]).
/// </summary>
public sealed class TsvRecord
{
    /// <summary>The spec QName for a TSV record element.</summary>
    public const string SpecQName = "tsv:record";

    /// <summary>The field values in this record (tab-separated).</summary>
    public IReadOnlyList<string> Fields { get; init; } = [];

    /// <summary>Number of fields in this record.</summary>
    public int FieldCount => Fields.Count;
}
